### PR TITLE
Ios picker

### PR DESCRIPTION
**Pull request by waynew99**
This branch is based on androidCreate on Wednesday after noon. It contains code in androidCreate that was not yet merged into main.

### CODE REVIEW

File: components/Code2/Code2.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import theme from "prism-react-renderer/themes/okaidia";
+import { ScrollView } from "react-native-gesture-handler";
+
+interface CodeProps {
+  code: string;
+  language: Language;
+  inEditor: boolean;
+}
+
+function Code2({ code, language, inEditor = false }: CodeProps) {
```
Comment (Nicholas-Sliter) [inline components/Code2/Code2.tsx:19]: We probably need a better name for code2

File: components/Code2/Code2.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  const courierFont =
+    Platform.OS === "ios" && parseInt(Platform.Version, 10) >= 15
+      ? "Courier New"
+      : "Courier";
+  const platformFont = Platform.OS === "ios" ? courierFont : "monospace";
+  const fontFamily = "Hack";
+
+
+  const rowStyle2 = (lineNumber: number) => {
+    //this function is now obsolete, refactor to simple style just in the stylesheet file
```
Comment (Nicholas-Sliter) [inline components/Code2/Code2.tsx:30]: do this

File: components/Code2/Code2.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                    <Text
+                      key={props.key}
+                      style={
+                        {
+                          color: props.style?.color ?? "#fff",
+                          fontFamily: fontFamily,
+                          fontSize: 14,
+                          lineHeight: 14,
+                          textAlignVertical: "top",
+                          //fontWeight: props.style?.fontWeight ?? "normal",
```
Comment (Nicholas-Sliter) [inline components/Code2/Code2.tsx:81]: remove

File: components/Code2/Code2.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                        } as any
+                      }
+                    >
+                      {!token.empty ? props.children : "\n" /* Not sure we need this*/ }
```
Comment (Nicholas-Sliter) [inline components/Code2/Code2.tsx:85]: we do

File: components/CodeEditor/CodeEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  return (
+    <ScrollView
+      style={styles.container}
+      contentContainerStyle={[styles.scrollContent, { height: height }]}
+      pointerEvents="box-none"
+    >
+      <TextInput
+        editable
+        //selection={{start: cursorPosition.start, end: cursorPosition.end}}
```
Comment (Nicholas-Sliter) [inline components/CodeEditor/CodeEditor.tsx:32]: remove

**Review by Nicholas-Sliter — CHANGES_REQUESTED** [review]

**Review by Cliffy2000 — COMMENTED** [review]
great work!

**Review by Cliffy2000 — APPROVED** [review]
actually an approve

**Review by Nicholas-Sliter — APPROVED** [review]
LGTM!

**Pull request merged**